presto-main/src/main/java/com/facebook/presto/server/QueryResource.java

```diff
@@ -84,6 +87,14 @@ public void cancelQuery(@PathParam("queryId") QueryId queryId)
         queryManager.cancelQuery(queryId);
     }
 
+    @PUT
+    @Path("{queryId}/kill")
```

Comment:
Why not use the path `kill/{queryId}`? I have seen in the past that ids of entities are usually at the end of the paths, but don't know whether we have or there are any conventions about that.

Reply:
Let's make this path `{queryId}/killed` since this is a resource (subresource / property of the query resource). Doing a PUT to the resource is setting the value of the `killed` resource (which contains the message / reason of why the query is killed).